src/Symfony/Component/DependencyInjection/ParameterBag/ParameterResolver.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\ParameterBag;
+
+use Symfony\Component\DependencyInjection\Exception\ParameterNotFoundException;
+use Symfony\Component\DependencyInjection\Exception\ParameterCircularReferenceException;
+use Symfony\Component\DependencyInjection\Exception\RuntimeException;
+
+/**
+ * Replaces parameter placeholders (%name%) by their values.
+ *
+ * @author Ener-Getick <egetick@gmail.com>
+ */
+final class ParameterResolver
```

Comment:
You may want to offer a different implementation. And locking this class with a final is a no0go too i think (this is not often done in Symfony's codebase). Or, if you realy want a final here, yo should at least offer a way to have somehting different (hence the interface)